pkg/bootstrapper/bootstrapper.go

```diff
@@ -537,48 +571,51 @@ func (wmcb *winNodeBootstrapper) controlService(cmd svc.Cmd, desiredState svc.St
 			return fmt.Errorf("timeout waiting for service to go to state=%d", desiredState)
 		}
 		time.Sleep(300 * time.Millisecond)
-		status, err = wmcb.kubeletSVC.Query()
+		status, err = wmcb.service.svcObj.Query()
 		if err != nil {
 			return fmt.Errorf("could not retrieve service status: %v", err)
 		}
 	}
 	return nil
 }
 
-// stopKubeletService stops the kubelet via the Windows service API
-func (wmcb *winNodeBootstrapper) stopKubeletService() error {
-	if wmcb.kubeletSVC == nil {
+// stopService stops the Windows service via the Windows service API
+func (wmcb *winNodeBootstrapper) stopService(serviceName string) error {
+	if wmcb.service.svcObj == nil {
 		return nil
 	}
 
-	kubeletIsRunning, err := wmcb.isKubeletServiceRunning()
+	isServiceRunning, err := wmcb.isServiceRunning()
 	if err != nil {
 		return fmt.Errorf("unable to check if kubelet service is running: %v", err)
 	}
-
-	if !kubeletIsRunning {
+	if !isServiceRunning {
 		return nil
 	}
 
-	return wmcb.controlService(svc.Stop, svc.Stopped)
+	if err := wmcb.controlService(svc.Stop, svc.Stopped); err != nil {
+		return fmt.Errorf("unable to stop Windows Service %s", serviceName)
+	}
+
+	return nil
 }
 
-// isKubeletServiceRunning returns true if the kubelet service is running
-func (wmcb *winNodeBootstrapper) isKubeletServiceRunning() (bool, error) {
-	status, err := wmcb.kubeletSVC.Query()
+// isServiceRunning returns true if the service is running
+func (wmcb *winNodeBootstrapper) isServiceRunning() (bool, error) {
```

Comment:
Instead of returning `bool` and `error` should we just return `error` if the service is not running and `nil` otherwise?

Reply:
the way we use this method is we check for the bool value and decide if there is anything to be done in case of stopping and starting a service, if the service is already in expected state we return nil. eg: If the service is not running and we want to stop it, we could just return nil as the variable value is false(returning an error because its not running doesn't fit here)